build-locally.sh

```diff
@@ -349,6 +349,8 @@ function build_generated_bom_pom {
     cd ${BASEDIR}/galasa-bom
 
     mvn \
+    --settings ${WORKSPACE_DIR}/obr/settings.xml \
+    -Dgpg.skip=true \
     -Dgpg.passphrase=${GPG_PASSPHRASE} \
```

Comment:
When the build-locally script runs, it still insists on `GPG_PASSPHRASE` being set - since we don't need to use GPG signing for local builds, we can remove that requirement.

For example, I get this error when `GPG_PASSPHRASE` isn't set and I try running the build-locally script:
```bash
[✖] Environment variable GPG_PASSPHRASE needs to be set.
[➜] Syntax: build-locally.sh [OPTIONS]
Options are:
-h | --help : Display this help text

Environment Variables:
SOURCE_MAVEN :
    Used to indicate where parts of the OBR can be obtained.
    Optional. Defaults to https://development.galasa.dev/main/maven-repo/managers/

LOGS_DIR :
    Controls where logs are placed.
    Optional. Defaults to creating a new temporary folder

GPG_PASSPHRASE :
    Mandatory.
    Controls how the obr is signed. Needs to be the alias of the private key of a
    public-private gpg pair. eg: For development you could use your signing github
    passphrase.
```